solutions/standard/variables.tf

```diff
@@ -0,0 +1,125 @@
+# Input Variables
+##############################################################################
+
+variable "ibmcloud_api_key" {
+  type        = string
+  description = "The IBM Cloud API key to deploy resources."
+  sensitive   = true
+}
+
+variable "provider_visibility" {
+  description = "Set the visibility value for the IBM terraform provider. Supported values are `public`, `private`, `public-and-private`. [Learn more](https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/guides/custom-service-endpoints)."
+  type        = string
+  default     = "private"
+
+  validation {
+    condition     = contains(["public", "private", "public-and-private"], var.provider_visibility)
+    error_message = "Invalid visibility option. Allowed values are 'public', 'private', or 'public-and-private'."
+  }
+}
+
+variable "use_existing_resource_group" {
+  type        = bool
+  description = "Whether to use an existing resource group."
+  default     = false
+}
+
+variable "resource_group_name" {
+  type        = string
+  description = "The name of a new or an existing resource group to provision the watsonx.data in. If a prefix input variable is specified, the prefix is added to the name in the `<prefix>-<name>` format."
+}
+
+variable "prefix" {
+  type        = string
+  description = "(Optional) Prefix to add to all resources created by this solution. To not use any prefix value, you can set this value to `null` or an empty string."
+  default     = "data-dev"
+}
+
+variable "name" {
+  type        = string
+  description = "The name of the watsonx.data instance. If a prefix input variable is specified, the prefix is added to the name in the `<prefix>-<name>` format."
+  default     = "dev"
+}
+
+variable "region" {
+  description = "The region where you want to deploy your instance."
+  type        = string
+  default     = "us-south"
+  validation {
+    condition     = local.kms_region == null || (local.kms_region == var.region)
+    error_message = "In case of enterprise plan, KMS instance should be in the same region as watsonx.data"
+  }
+}
+
+variable "plan" {
+  type        = string
+  description = "The plan that is required to provision the watsonx.data instance. Possible values are: essentials, standard. [Learn more](https://www.ibm.com/products/watsonx-data/pricing)."
+  default     = "lakehouse-enterprise"
+}
+
+variable "resource_tags" {
+  type        = list(string)
+  description = "Optional list of tags to describe the newly created watsonx.data instance."
+  default     = []
+}
+
+variable "access_tags" {
+  type        = list(string)
+  description = "A list of access tags to apply to the watsonx.data instance. [Learn more](https://cloud.ibm.com/docs/account?topic=account-access-tags-tutorial)."
+  default     = []
+}
+
+variable "enable_kms_encryption" {
+  description = "Flag to enable KMS encryption. If set to true, a value must be passed for `existing_kms_key_crn`. This is applicable only for Enterprise plan."
```

Comment:
If set to true, a value must be passed for `existing_kms_key_crn` -> it should include existing_kms_instance_crn too

Reply:
Done